### PR TITLE
Setup for Managing Global Streams, Daily Challenges Page

### PR DESCRIPTION
**Pull request by Sudhindra3000**
### Features:
- Setup for managing Global streams
- DailyChallenges page with challenges for each day in a tab layout
  - Each `SingleDayChallenges` has it's own state for fetching and showing challenges
  - All pages greater than current marketday are locked
- Message is shown when daily challenges is closed
  - `isDailyChallengesOpen` is updated using `GameState` stream

### Todo:
- UI in DailyChallenges page
- Retry button in Splash page when state is `DalalLoginFailed`
- Realtime updates in progress of Challenges using stockListMap

### CODE REVIEW

File: lib/blocs/auth/login/login_cubit.dart [hunk 1]
```diff
@@ -18,14 +21,39 @@ class LoginCubit extends Cubit<LoginState> {
...
-        emit(LoginSuccess(resp));
-        userBloc.add(UserLogIn(resp));
-      } else {
-        emit(LoginFailure(resp.statusMessage));
+      final sessionId = loginResp.sessionId;
+      if (loginResp.statusCode != LoginResponse_StatusCode.OK) {
+        emit(LoginFailure(loginResp.statusMessage));
+        return;
       }
+      final stockResponse = await actionClient.getStockList(
```
Comment (satyamurti) [inline lib/blocs/auth/login/login_cubit.dart:31]: I don't think calling these many rpc calls in login cubit is good. Login Cubit should be responsible for only Logging in the User.

File: lib/blocs/auth/login/login_cubit.dart [hunk 2]
```diff
@@ -18,14 +21,39 @@ class LoginCubit extends Cubit<LoginState> {
...
+        SubscribeRequest(dataStreamType: DataStreamType.GAME_STATE),
+        options: sessionOptions(sessionId),
+      );
+      if (gameStateResp.statusCode != SubscribeResponse_StatusCode.OK) {
+        emit(LoginFailure(gameStateResp.statusMessage));
+        return;
+      }
+      final gameStateStream = streamClient.getGameStateUpdates(
+        gameStateResp.subscriptionId,
+        options: sessionOptions(sessionId),
```
Comment (satyamurti) [inline lib/blocs/auth/login/login_cubit.dart:49]: How about we have a separate cubits for these are update loading bar and text showing what is current state of the app in UI according to the calls , similar to popular games like league of legends , PUBG , Valorant.

Comment (satyamurti) [reply lib/blocs/auth/login/login_cubit.dart:49]: image
image
image
image

Comment (satyamurti) [reply lib/blocs/auth/login/login_cubit.dart:49]: Like this we can show in UI

File: lib/blocs/daily_challenges/daily_challenges_page_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      }
+    } catch (e) {
+      logger.e(e);
+      emit(const DailyChallengesPageFailure(failedToReachServer));
+    }
+  }
+
+  /// Updates `isDailyChallengeOpen` in [DailyChallengesPageSuccess]
+  /// Only call this after [getChallengesConfig] is succesful
+  Future<void> listenToGameStateSteam() async {
```
Comment (satyamurti) [inline lib/blocs/daily_challenges/daily_challenges_page_cubit.dart:39]: Steam is typo?

**Review by satyamurti — CHANGES_REQUESTED** [review]

File: lib/blocs/auth/login/login_cubit.dart (hunk 2 above)
Comment (satyamurti) [reply lib/blocs/auth/login/login_cubit.dart:49]: This is League of Legends

File: lib/blocs/daily_challenges/daily_challenges_page_cubit.dart (hunk 3 above)
Comment (Sudhindra3000) [reply lib/blocs/daily_challenges/daily_challenges_page_cubit.dart:39]: ya i will change

**Review by satyamurti — APPROVED** [review]

File: lib/blocs/auth/login/login_cubit.dart (hunk 2 above)
Comment (Sudhindra3000) [reply lib/blocs/auth/login/login_cubit.dart]: Nice idea. but it wont look as appealing in our case because all the calls are completed pretty fast

Comment (Sudhindra3000) [reply lib/blocs/auth/login/login_cubit.dart]: Atleast that is the case now. If we add more streams and other stuff then it might take some time

Comment (satyamurti) [reply lib/blocs/auth/login/login_cubit.dart]: okay fine

File: lib/pages/daily_challenges/daily_challenges_page.dart [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final days = 1.to(widget.successState.totalMarketDays);
+    return Column(
```
Comment (satyamurti) [inline lib/pages/daily_challenges/daily_challenges_page.dart:85]: Use `Responsive` widget here. Refer to homepage.

**Review by satyamurti — CHANGES_REQUESTED** [review]

File: lib/pages/daily_challenges/daily_challenges_page.dart (hunk 4 above)
Comment (Sudhindra3000) [reply lib/pages/daily_challenges/daily_challenges_page.dart:86]: Ok

Comment (Sudhindra3000) [reply lib/pages/daily_challenges/daily_challenges_page.dart:86]: Actually i'm doing a separate pr for UI. I will do this there

**Review by satyamurti — APPROVED** [review]

**Pull request merged**